subcontracted_service/models/procurement.py

```diff
@@ -31,6 +31,9 @@ def _assign(self):
         if not res:
             rule_id = self._find_suitable_rule()
             if rule_id:
-                self.write({'rule_id': rule_id})
```

Comment:
change to self.write({'rule_id': rule_id.id})

Reply:
remove last .id like:
self.warehouse_id.subcontracting_service_proc_rule_id

Reply:
I think this way it should work, could you try in your instance?

Reply:
Yes, it works.
i've just leave the old self.write({'rule_id': rule_id}) in case it come as object.